modules/fluentd_new/modules/fluentd_base/variables.tf

```diff
@@ -0,0 +1,227 @@
+
+variable "tenancy" {
+  description = "The tenancy of the instance. Must be one of: default or dedicated."
+  type        = string
+  default     = "default"
+}
+
+variable "root_volume_ebs_optimized" {
+  description = "If true, the launched EC2 instance will be EBS-optimized."
+  type        = bool
+  default     = false
+}
+
+variable "root_volume_type" {
+  description = "The type of volume. Must be one of: standard, gp2, or io1."
+  type        = string
+  default     = "standard"
+}
+
+variable "root_volume_size" {
+  description = "The size, in GB, of the root EBS volume."
+  type        = number
+  default     = 50
+}
+
+variable "root_volume_delete_on_termination" {
+  description = "Whether the volume should be destroyed on instance termination."
+  default     = true
+  type        = bool
+}
+
+variable "wait_for_capacity_timeout" {
+  description = "A maximum duration that Terraform should wait for ASG instances to be healthy before timing out. Setting this to '0' causes Terraform to skip all Capacity Waiting behavior."
+  type        = string
+  default     = "10m"
+}
+
+variable "health_check_type" {
+  description = "Controls how health checking is done. Must be one of EC2 or ELB."
+  type        = string
+  default     = "EC2"
+}
+
+variable "health_check_grace_period" {
+  description = "Time, in seconds, after instance comes into service before checking health."
+  type        = number
+  default     = 300
+}
+
+variable "instance_profile_path" {
+  description = "Path in which to create the IAM instance profile."
+  type        = string
+  default     = "/"
+}
+
+variable "http_port" {
+  description = "The port to use for HTTP"
+  type        = number
+  default     = 4646
+}
+
+variable "rpc_port" {
+  description = "The port to use for RPC"
+  type        = number
+  default     = 4647
+}
+
+variable "serf_port" {
+  description = "The port to use for Serf"
+  type        = number
+  default     = 4648
+}
+
+variable "ssh_port" {
+  description = "The port used for SSH connections"
+  type        = number
+  default     = 22
+}
+
+variable "security_groups" {
+  description = "Additional security groups to attach to the EC2 instances"
+  type        = list(string)
+  default     = []
+}
+
+variable "tags" {
+  description = "List of extra tag blocks added to the autoscaling group configuration. Each element in the list is a map containing keys 'key', 'value', and 'propagate_at_launch' mapped to the respective values."
+  type = list(object({
+    key                 = string
+    value               = string
+    propagate_at_launch = bool
+  }))
+  default = []
+
+}
+
+variable "ebs_block_devices" {
+  description = "List of ebs volume definitions for those ebs_volumes that should be added to the instances created with the EC2 launch-configuration. Each element in the list is a map containing keys defined for ebs_block_device (see: https://www.terraform.io/docs/providers/aws/r/launch_configuration.html#ebs_block_device."
```

Comment:
Nit: The description is far too lengthy and this affects the doc generation.
Consider split the second sentence into a comment instead?